frontend/src/app/_models/menu.ts

```diff
@@ -1,11 +1,13 @@
+import {NavigationRoute} from './route';
+
 
 export interface MenuHierarchy {
   name: string;
   label: string;
-  type: 'Project' | 'Node' | 'Branch' | 'Directory' | 'Page';
+  type: string; // 'Node' | 'Project' | 'Branch' | 'Directory' | 'Page';
```

Comment:
The union type `'A' | 'B'` is quite powerful, it helps spot places where you use an unexpected value. But maybe it doesn't makes sense anymore with the changes and you can ignore my comment ;)

Reply:
it's breaking the compilation, i don't understand why.... see the same remark done by  Lorie ;)

Reply:
Got it.
With a specific type, it's better :

> type MenuType = 'Node' | 'Project' | 'Branch' | 'Directory' | 'Page';

> export interface MenuHierarchy {
>   name: string;
>   label: string;
>   type:  MenuType;